libs/pages/application/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.tsx

```diff
@@ -0,0 +1,98 @@
+import { ApplicationAdvancedSettings } from 'qovery-typescript-axios'
+import { useEffect, useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router'
+import {
+  editApplicationAdvancedSettings,
+  fetchApplicationAdvancedSettings,
+  fetchDefaultApplicationAdvancedSettings,
+  getApplicationsState,
+  selectApplicationById,
+} from '@console/domains/application'
+import { ApplicationEntity } from '@console/shared/interfaces'
+import { objectFlattener } from '@console/shared/utils'
+import { AppDispatch, RootState } from '@console/store/data'
+import PageSettingsAdvanced from '../../ui/page-settings-advanced/page-settings-advanced'
+import { initFormValues } from './utils'
+
+export function PageSettingsAdvancedFeature() {
+  const { applicationId = '' } = useParams()
+
+  const application = useSelector<RootState, ApplicationEntity | undefined>(
+    (state) => selectApplicationById(state, applicationId),
+    (a, b) => {
+      return a?.id === b?.id && a?.advanced_settings?.loadingStatus === b?.advanced_settings?.loadingStatus
+    }
+  )
+  const defaultSettings = useSelector<RootState, ApplicationAdvancedSettings | undefined>(
+    (state) => getApplicationsState(state).defaultApplicationAdvancedSettings.settings
+  )
+  const [keys, setKeys] = useState<string[]>([])
+
+  const dispatch = useDispatch<AppDispatch>()
+  const methods = useForm({ mode: 'onChange' })
+
+  // at the init fetch the default settings advanced settings
+  useEffect(() => {
+    dispatch(fetchDefaultApplicationAdvancedSettings())
+  }, [dispatch])
+
+  // when application is ready, and advanced setting has never been fetched before
+  useEffect(() => {
+    if (application && !application.advanced_settings?.loadingStatus) {
+      dispatch(fetchApplicationAdvancedSettings({ applicationId }))
+    }
+  }, [dispatch, application, applicationId])
+
+  // init the keys when application is updated
+  useEffect(() => {
+    if (application) {
+      if (
+        application.advanced_settings?.current_settings &&
+        application.advanced_settings?.loadingStatus === 'loaded'
+      ) {
+        setKeys(Object.keys(application.advanced_settings.current_settings).sort())
+      }
+    }
+  }, [application])
+
+  // init form
+  useEffect(() => {
+    if (application && application.advanced_settings?.loadingStatus === 'loaded') {
+      methods.reset(initFormValues(keys, application))
+    }
+  }, [application, keys, methods])
+
+  const onSubmit = methods.handleSubmit((data) => {
```

Comment:
Do you think it could be nice to test this function? because you remove some keys?

Reply:
This is exactly what I was saying you in your pr. Instead of testing this function, I'm just making sure that the `editApplicationAdvancedSettings` function has been called with the correct payload. 
In the end it allows us to test the handleSubmit without actually mocking the submit which can be painful